### DISCUSSION TITLE
Implement Threshold Signature scheme

### DISCUSSION THREAD

**Discussion by pool2win** [braidpool/braidpool#166] [Ideas]
@pool2win has done a lot of work on a FROST Federation, which could be used to custody coinbase rewards across multiple blocks (Rolling Coinbase Aggregation RCA), and then pay out using the Unspent Hasher Payout Object (UHPO) proposal and Eltoo. [His work is here.](https://github.com/pool2win/frost-federation)

This idea is not safe if Braidpool has less than 34% of the network hashrate, because a malicious miner who can win 67% of the shares can also become the entire quorum of signing members for the custody updates, and this allows him to steal all funds. If/when Braidpool ever reaches 34% of the network, I may resurrect this idea. But in the meantime we're going with PPLNS and actively researching other options.

-----

Use zcash' FROST implementation https://github.com/ZcashFoundation/frost

Decisions on DKG and TSS here:

https://blog.opdup.com/2023/08/22/frost-for-braidpool.html

Use a mock clock for rounds - wait for N messages.

**Comment by average-gary** [comment]
https://github.com/m1sterc001guy/roastr
This uses FROST/ROAST for Schnoor signatures over nostr events using AlephBFT for consensus when signing.

**Comment by pool2win** [comment]
Nice. Thanks for the pointer. Will take a look.

QQ @marathon-gary - is there is a design / architecture overview of the work? How do the meet the network model requirements for FROST when providing the broadcast and the unicast channels between federation members?

I distilled the requirements here https://blog.opdup.com/development-updates/2024/05/04/p2p-channels-for-frost.html

**Comment by average-gary** [comment]
> QQ @marathon-gary - is there is a design / architecture overview of the work? How do the meet the network model requirements for FROST when providing the broadcast and the unicast channels between federation members?

I am only aware of the reference implementation but will try to summarize. Fedimint assumes a minimum threshold of connectivity between the Guardian (nodes/servers) for consensus to be functional. This means that a FROST quorum with similar (or greater) threshold should always be able to sign, assuming the Fedimint in general is functioning as expected.

If you're inclined, you may be able to chat with m1sterc001guy directly in the [Fedimint Discord](https://chat.fedimint.org)

**Comment by pool2win** [comment]
Thanks. I'll connect with m1sterc001guy on the discord.
